YubiKit/YubiKit/Sessions/Shared/Services/OATH/YKFKeyOATHService.m

```diff
@@ -112,6 +112,21 @@ - (void)executeDeleteRequest:(YKFKeyOATHDeleteRequest *)request completion:(YKFK
     }];
 }
 
+- (void)executeRenameRequest:(YKFKeyOATHRenameRequest *)request completion:(YKFKeyOATHServiceCompletionBlock)completion {
+    YKFParameterAssertReturn(request);
+    YKFParameterAssertReturn(completion);
+    
+    YKFKeySessionError *credentialError = [YKFOATHCredentialValidator validateCredential:request.credential includeSecret:NO];
```

Comment:
You seem to validate current credential that was passed, but not validating passed new account name and new issuer name?

Reply:
Fixed. We now use the same logic as when creating a new credential.

Reply:
You mean that you wanted to check renamedCredential here?

Reply:
I added the check in `YKFKeyOATHRenameRequest` in the same way we do for the put request but I don't think it's very good. Now you have to handle potential errors in two places. When creating a request it might return `nil` if there's an error in the input *and* when you execute the request if the `YKFOATHCredentialValidator` fails to validate the credential. I would prefer if the initializers of our requests would not be failable and all error handling would take place in the execution of the request by the `YKFOATHCredentialValidator`. What do you think?

Reply:
Where do you check that new name is not out of the length boundaries? 
I don't think that returning nil from request is good idea (unless passed arguments are nil or developer messed something up). You won't be able propagate why operation failed. More over you operation silently fail (because you're using asserts), while you need to return completion with error in case of wrong user input.
Asserts for developers
Errors for developers and potentially users

Reply:
I gave up when I realised validation is taking place in a lot of different places. If we look at the put request we first validate that the credential has a lot of properties when we *initialize* the `YKFKeyOATHPutRequest`. This is done by macros that will halt execution in debug builds and return `nil` in production. As you point out no information as to why it failed is passed to the consumer of the SDK. Second we validate that the length of the key and secret is not bigger than specified when we *execute* the request. This error will be sent to the call site via the completion block.

What I would like to do is change the initializer of all requests to return `nonnull` and handle all validation in the `YKFOATHCredentialValidator` and in the execute methods. That way we could send meaningful error codes when we try to execute the request for *all* errors that can happen. Another benefit of this is that  we don't have to handle the failable initialization of the requests. Today this is done by force unwrapping in the Authenticator app and it would be nice if we didn't have to do that. What's your thought on this approach?

Reply:
Overall I like the idea of having validation in one place. 
But I think rewriting all of requests is big refactoring (potentially breaking external customers). I'm not sure that benefit worth it.But it's your call.
For this change I would just check for expected errors and return some custom error code for them (Key will return some error code for all of these scenarios, but user needs to know what each code mean)
- User didn't change the name, 
- New name out of bounds (longer than key supports)
- Operation is not supported
You don't need to check if new name or label is empty (key actually supports it)
And that's it.
There are some other expected errors. Such as:
- name already exists
- credential was already removed
Not sure that it's appropriate to check it within SDK. And we didn't do that for other operations. So may be it's ok to delegate handling of these scenarios to user.

Reply:
Yes, I agree this PR is not the place to do that refactoring. I'll add some basic validation in the execution call similar to what we do for a put request and call it a day. Will probably revisit error handling in a later PR. Thanks for the comments!